platform/kube/controller.go

```diff
@@ -298,6 +399,22 @@ func (c *Controller) List(kind, namespace string) (map[model.Key]proto.Message,
 	return out, errs
 }
 
+func (c *Controller) listIngresses(kind, namespace string) (map[model.Key]proto.Message, error) {
```

Comment:
What is it that's not needed here? The `listIngresses` method gets called from `Controller.List(kind, namespace)` according to the actual kind. In practice, this is being used by the envoy's ingress watcher.